### PR TITLE
Cleanup old components, add FillGradient and InteractiveLayer

### PR DESCRIPTION
**Pull request by flehoux**
Kapture 2019-04-26 at 14 32 46

### CODE REVIEW

File: src/Sparklines.js [hunk 1]
```diff
@@ -1,65 +1,97 @@
...
+    preserveAspectRatio: "none", //https://www.w3.org/TR/SVG/coords.html#PreserveAspectRatioAttribute
+    margin: 2
+  };
 
-    constructor (props) {
-        super(props);
-    }
+  constructor(props) {
+    super(props);
+  }
```
Comment (wfortin) [inline src/Sparklines.js]: Remove unneeded constructor.

File: src/Sparklines.js [hunk 2]
```diff
@@ -1,65 +1,97 @@
...
+      svgHeight,
+      preserveAspectRatio,
+      margin,
+      style,
+      max,
+      min
+    } = this.props;
 
-        if (data.length === 0) return null;
+    if (data.length === 0) return null;
```
Comment (wfortin) [inline src/Sparklines.js]: Use brackets in `if` statements.

File: src/Sparklines.js [hunk 3]
```diff
@@ -1,65 +1,97 @@
...
-        const svgOpts = { style: style, viewBox: `0 0 ${width} ${height}`, preserveAspectRatio: preserveAspectRatio };
-        if (svgWidth > 0) svgOpts.width = svgWidth;
-        if (svgHeight > 0) svgOpts.height = svgHeight;
+    const svgOpts = {
+      style: style,
+      viewBox: `0 0 ${width} ${height}`,
+      preserveAspectRatio: preserveAspectRatio
+    };
+    if (svgWidth > 0) svgOpts.width = svgWidth;
+    if (svgHeight > 0) svgOpts.height = svgHeight;
```
Comment (wfortin) [inline src/Sparklines.js]: also bracketsHere.

File: src/SparklinesInteractiveLayer.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    onMouseMove: PropTypes.func,
+    onMouseLeave: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      cx: OUT_OF_BOUND,
+      cy: OUT_OF_BOUND
+    };
```
Comment (wfortin) [inline src/SparklinesInteractiveLayer.js]: remove constructor and set the state in the class directly.

**Comment by flehoux** [comment]
Make sense, should we also take some time and typescriptify this thing at some point?

**Comment by wfortin** [comment]
Would definitely be nice yeah.

**Pull request merged**